src/CodeGen_LLVM.cpp

```diff
@@ -629,6 +629,19 @@ void CodeGen_LLVM::end_func(const std::vector<LoweredArgument>& args) {
     // Generate the function declaration and argument unpacking code.
     begin_func(f.linkage, simple_name, extern_name, f.args);
 
+    // If building with MSAN, ensure that calls to halide_msan_annotate_buffer_is_initialized()
+    // happen for every output buffer if the function succeeds.
+    if (f.linkage == LoweredFunc::External && target.has_feature(Target::MSAN)) {
```

Comment:
Yeah, what's more, we really want to mark every realized buffer anyway, even if it's not an output: it might be seen by an extern stage. Simpler to mark everything.